packages/ui/src/components/cart/CartPopover.tsx

```diff
@@ -443,19 +454,25 @@ export function CartPopover({
                   (transaction?.status === CheckoutStatus.Idle ||
                     !displayPendingTransaction) && (
                     <Button
-                      disabled={!hasEnoughCurrency}
+                      disabled={!hasEnoughCurrency && isConnected}
                       onClick={async () => {
-                        checkout()
-                          .then(() => {
-                            setDisplayPendingTransaction(true)
-                          })
-                          .catch((e) => {
-                            console.error(e)
-                            setDisplayPendingTransaction(false)
-                          })
+                        if (!isConnected) {
+                          onConnectWallet?.()
```

Comment:
I think we should also close the cart popover here. There are some mouse event conflicts when both the cart popover and rainbowkit modals are open

Reply:
Actually, now realizing developers can handle that themselves